packages/components/src/tab-panel/types.ts

```diff
@@ -34,7 +35,7 @@ type Tab< IconProps = unknown > = {
 export type TabButtonProps< IconProps = unknown > = {
 	children: ReactNode;
 	label?: string;
-	onClick: ( event: MouseEvent ) => void;
+	onClick: Parameters< typeof Button >[ 0 ][ 'onClick' ];
```

Comment:
Corrects a TS error:

```
packages/components/src/tab-panel/index.tsx:26:3 - error TS2322: Type '{ children: ReactNode; label?: string | undefined; onClick: (event: MouseEvent) => void; showTooltip?: boolean | undefined; className?: string | undefined; icon?: IconType<...> | undefined; ... 5 more ...; __experimentalIsFocusable: true; }' is not assignable to type 'Omit<Pick<DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, "key" | keyof ButtonHTMLAttributes<...>> | Pick<...>, "as" | keyof ButtonProps>'.
  Types of property 'onClick' are incompatible.
    Type '(event: MouseEvent) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement> | MouseEventHandler<HTMLAnchorElement> | undefined'.
      Type '(event: MouseEvent) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
        Types of parameters 'event' and 'event' are incompatible.
          Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is missing the following properties from type 'MouseEvent': offsetX, offsetY, x, y, and 14 more.
```

Uses the `Parameters` generic instead of `WordPressComponentProps< ButtonProps, TagName >[ 'onClick' ]`